modules/ingest-common/src/test/java/org/elasticsearch/ingest/common/ForEachProcessorTests.java

```diff
@@ -270,4 +273,26 @@ public void testNestedForEach() throws Exception {
         assertThat(result.get(1), equalTo("JKL"));
     }
 
+    public void testIgnoreMissing() throws Exception {
+        IngestDocument originalIngestDocument = new IngestDocument(
+            "_index", "_type", "_id", null, null, null, Collections.emptyMap()
+        );
+        IngestDocument ingestDocument = new IngestDocument(
+            "_index", "_type", "_id", null, null, null, Collections.emptyMap()
+        );
+        ForEachProcessor processor = new ForEachProcessor("_tag", "_ingest._value", new AbstractProcessor("noop") {
```

Comment:
just so we are more explicit, maybe it would be clearer to re-use the test-processor to validate that things 
are not executed?

maybe something like this

```
TestProcessor testProcessor = new TestProcessor(doc -> {});
ForEachProcessor processor = new ForEachProcessor("_tag", "_ingest._value", testProcessor);
processor.execute(ingestDocument);
assertThat(testProcessor.getInvokedCounter(), equalTo(0));
```